### DISCUSSION TITLE
Support for semantic version strings and comparison

### DISCUSSION THREAD

**Discussion by cueckoo** [cue-lang/cue#912] [Q&A] [answered]
_Originally opened by @codewithcheese in https://github.com/cuelang/cue/discussions/912_

Hi CUEbies

Do you know of any implementations for semantic version and comparison in-built or CUE? I see there is an issue from a year ago here: https://github.com/cuelang/cue/issues/407

I am about to implement a CUE based solution for semantic version strings and comparison but double checking before coding.

I'm planning a CUE based solution, use regexp to parse a version string into its components something like `{ major: number, minor: number, patch: number, extension:string }` then implement a comparison function; a long list of conditionals which probably wont be pretty but get the job done.

We use semantic version strings in our config files to represent versions and also have a need for version comparison to check that component updates are mutually compatible.

**Comment by cueckoo** [comment, marked as answer]
_Original reply by @myitcv in https://github.com/cuelang/cue/discussions/912#discussioncomment-792346_

Apologies for the delay in replying, @codewithcheese. A regexp approach will work (per https://github.com/cuelang/cue/issues/407#issuecomment-760496843). And per @mpvl the plan at the moment is not necessarily to provide a builtin equivalent of https://pkg.go.dev/golang.org/x/mod/semver, rather to consider what building blocks would be better primitive (@mpvl gives the example of https://pkg.go.dev/golang.org/x/text/collate

**Reply by mpvl** [reply]
^ this.
